src/ngircd/irc-login.c

```diff
@@ -259,6 +260,22 @@ IRC_NICK( CLIENT *Client, REQUEST *Req )
 				Client_SetType( Client, CLIENT_GOTNICK );
 		} else {
 			/* Nickname change */
+
+			/* Check that the user isn't on any channels set +N */
+			if(Client_Type(Client) == CLIENT_USER &&
+			   !Client_HasMode(Client, 'o')) {
+				chan = Channel_First();
+				while (chan) {
+					if(Channel_IsMemberOf(chan, Client) &&
+					   Channel_HasMode(chan, 'N'))
```

Comment:
Probably switch these two lines? So that we check first if the channel is +N, and only if this is so have to check if the client is a member of this channel? This should perform better because we don't have to walk an other list.